### PR TITLE
tests: drivers: spi: spi_error_cases: Enable test execution on nrf54h20

### PR DESCRIPTION
**Pull request by nordic-segl**
Add overlay file for nrf54h20.
Align code and DTS between spi_error_cases and spi_controller_peripheral.

### CODE REVIEW

**Comment by nordic-segl** [comment]
Tested locally - works correctly on nrf54l15pdk and nrf54h20dk.

**Comment by nordic-segl** [comment]
@nordic-krch, @anangl, @nordic-piks: Could You please take part in the review?

**Review by nordic-piks — APPROVED** [review]

File: tests/drivers/spi/spi_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/delete-node/ &exmif;
```
Comment (nordic-piks) [inline tests/drivers/spi/spi_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:7]: Maybe it is worth to mention that this line is here only due to https://github.com/zephyrproject-rtos/zephyr/issues/72657

Comment (nordic-segl) [reply tests/drivers/spi/spi_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:8]: I added a comment

**Review by nordic-krch — DISMISSED** [review]

File: tests/drivers/spi/spi_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay [hunk 2]
```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Workaround for https://github.com/zephyrproject-rtos/zephyr/issues/72657 */
```
Comment (nordic-segl) [inline tests/drivers/spi/spi_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:7]: I sow it was fixed.

**Comment by nordic-segl** [comment]
[rebased on top of main]

**Comment by nordic-segl** [comment]
Dear reviewers, please find some time to do the review.

**Review by nordic-krch — DISMISSED** [review]

File: tests/drivers/spi/spi_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+&gpio0 {
+	status = "okay";
+};
+
+&gpiote130 {
+	status = "okay";
+	owned-channels = <7>;
+};
+
+dut_spi: &spi130 {
```
Comment (anangl) [inline tests/drivers/spi/spi_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:55]: This node label is no longer used.

File: tests/drivers/spi/spi_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpiote130 {
+	status = "okay";
+	owned-channels = <7>;
+};
```
Comment (anangl) [inline tests/drivers/spi/spi_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:53]: Is this actually needed for this test?

Comment (nordic-segl) [reply tests/drivers/spi/spi_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:53]: Right, copy-paste error. I will remove it.

File: tests/drivers/spi/spi_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay (hunk 3 above)
Comment (nordic-segl) [reply tests/drivers/spi/spi_error_cases/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:55]: I will remove it.

**Review by anangl — APPROVED** [review]

**Review by nordic-krch — APPROVED** [review]

**Comment by nordic-segl** [comment]
[rebased on top of main]

**Comment by nordic-segl** [comment]
@tbursztyka Could You take a look?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Please use inclusive language in the Git commit message. See https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#rule-a-2-inclusive-language for more information.

**Review by fabiobaltieri — APPROVED** [review]

**Comment by nordic-segl** [comment]
I removed inclusive language from the commit message.

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by carlescufi** [comment]
@nordic-segl can you please rebase this? there seems to be a conflict

**Comment by nordic-segl** [comment]
> @nordic-segl can you please rebase this? there seems to be a conflict

Rebased. This might be other issue because I didn't have to resolve any conflicts.

**Comment by nordic-segl** [comment]
tested locally on nrf54h20 - test is passing.

./zephyr/scripts/twister --ninja --inline-logs --jobs 1 -T zephyr/tests/drivers/spi/spi_error_cases/ --platform nrf54h20dk/nrf54h20/cpuapp --device-testing --device-serial /dev/serial/by-id/usb-SEGGER_J-Link_001051114155-if00 --no-clean -v -v --fixture gpio_spi_loopback

**Comment by nordic-segl** [comment]
Tested locally on nrf54l15pdk - test is passing.

./zephyr/scripts/twister --ninja --inline-logs --jobs 1 -T zephyr/tests/drivers/spi/spi_error_cases/ --platform nrf54l15pdk/nrf54l15/cpuapp --device-testing --device-serial /dev/serial/by-id/usb-SEGGER_J-Link_001057750650-if02 --no-clean -v -v --fixture gpio_spi_loopback

**Pull request merged**